weather-oracle/sources/weather.move

```diff
@@ -0,0 +1,377 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This module defines a weather oracle that can post weather updates for major cities around the world.
+module oracle::weather {
+    use std::option::{Self, Option};
+    use std::string::{Self, String};
+
+    use sui::dynamic_object_field as dof;
+    use sui::object::{Self, UID};
+    use sui::package;
+    use sui::transfer::{Self};
+    use sui::tx_context::{Self, TxContext};
+
+    /// Define a capability for the admin of the oracle.
+    struct AdminCap has key, store { id: UID }
+
+    /// // Define a one-time witness to create the `Publisher` of the oracle.
+    struct WEATHER has drop {}
+
+    // Define a struct for the weather oracle
+    struct WeatherOracle has key {
+        id: UID,
+        /// The address of the oracle.
+        address: address,
+        /// The name of the oracle.
+        name: String,
+        /// The description of the oracle.
+        description: String,
+    }
+
+    // Define a struct for each city that the oracle covers
+    struct CityWeatherOracle has key, store {
+        id: UID,
+        geoname_id: u32, // The unique identifier of the city
```

Comment:
How important are these properties and is there a value to have them as fields (as opposed to storing a VecMap)? No strong opinion about this one but I do feel like metadata is distracting from the main parameters (what if some field is changed...), and the amount of getters needed could be trimmed to `read(): &VecMap<String, String>`

Reply:
Thank you very much @damirka for the feedback. I working on a refactored version, so if we change them to `VecMap ` it is clear that the contracts will be smaller. Except from that what else do we gain?